stable/database/templates/statefulset.yaml

```diff
@@ -315,6 +317,10 @@ spec:
         matchLabels:
           database: {{ .Values.database.name }}
 {{- end }}
+      {{- if .Values.database.persistence.dataSourceRef.name }}
+      dataSourceRef:
+        {{- toYaml .Values.database.persistence.dataSourceRef | nindent 8}}
+      {{- end }}
```

Comment:
I think we may want to allow the `dataSourceRef` to be optional as well, to be compatible with versions of `values.yaml`. That can be implemented by wrapping a second conditional around this as follows:

```
      {{- if .Values.database.persistence.dataSourceRef }}
      {{- if .Values.database.persistence.dataSourceRef.name }}
      dataSourceRef:
        {{- toYaml .Values.database.persistence.dataSourceRef | nindent 8}}
      {{- end }}
      {{- end }}
```

Reply:
Good idea, I will fix the logic and add a template test for the scenario.